src/crimson/admin/admin_socket.cc

```diff
@@ -44,25 +44,14 @@ tell_result_t::tell_result_t(std::unique_ptr<Formatter> formatter)
   formatter->flush(out);
 }
 
-seastar::future<>
-AdminSocket::register_command(std::unique_ptr<AdminSocketHook>&& hook)
+void AdminSocket::register_command(std::unique_ptr<AdminSocketHook>&& hook)
 {
-  return seastar::with_lock(servers_tbl_rwlock,
```

Comment:
This simplification looks sane. I would just write the requirement (no late registration) down.

Reply:
i think actually it's fine to register hooks after the admin sock infra starts. as all the access to the `hooks` are synchronous. and the `hook` is valid after the `hooks` is mutated.